### ISSUE TITLE
Twisted integration fails

### ISSUE THREAD

**Issue by zgoda-mobica** [rollbar/pyrollbar#104]
Simplest possible Twisted code:

```
from twisted.internet import reactor
import rollbar

_ACCESS_TOKEN = 'my access token'
_ENV = 'framework'

def main():
    rollbar.init(_ACCESS_TOKEN, environment=_ENV, handler='twisted')
    rollbar.report_message('test message from Twisted')
    reactor.run()

if __name__ == '__main__':
    main()
```

This fails with:

```
Unhandled error in Deferred:

Traceback (most recent call last):
  File "/home/jazg/v/twt/lib/python2.7/site-packages/rollbar/__init__.py", line 655, in _report_exc_info
    send_payload(payload, data.get('access_token'))
  File "/home/jazg/v/twt/lib/python2.7/site-packages/rollbar/__init__.py", line 456, in send_payload
    _send_payload_twisted(payload, access_token)
  File "/home/jazg/v/twt/lib/python2.7/site-packages/rollbar/__init__.py", line 1244, in _send_payload_twisted
    _post_api_twisted('item/', payload, access_token=access_token)
  File "/home/jazg/v/twt/lib/python2.7/site-packages/twisted/internet/defer.py", line 1274, in unwindGenerator
    return _inlineCallbacks(None, gen, Deferred())
--- <exception caught here> ---
  File "/home/jazg/v/twt/lib/python2.7/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "/home/jazg/v/twt/lib/python2.7/site-packages/rollbar/__init__.py", line 1263, in _post_api_twisted
    StringProducer(payload))
  File "/home/jazg/v/twt/lib/python2.7/site-packages/twisted/web/client.py", line 1592, in request
    parsedURI = URI.fromBytes(uri)
  File "/home/jazg/v/twt/lib/python2.7/site-packages/twisted/web/client.py", line 629, in fromBytes
    scheme, netloc, path, params, query, fragment = http.urlparse(uri)
  File "/home/jazg/v/twt/lib/python2.7/site-packages/twisted/web/http.py", line 166, in urlparse
    raise TypeError("url must be bytes, not unicode")
exceptions.TypeError: url must be bytes, not unicode
WARNING:rollbar:Traceback frame not a types.FrameType. Ignoring.
```

And program enters infinite loop.

Environment:

```
$ pip list
cffi (1.5.2)
cryptography (1.3.1)
decorator (4.0.9)
enum34 (1.1.2)
idna (2.1)
ipaddress (1.0.16)
ipdb (0.9.0)
ipython (4.1.2)
ipython-genutils (0.1.0)
path.py (8.1.2)
pexpect (4.0.1)
pickleshare (0.6)
pip (8.1.1)
ptyprocess (0.5.1)
pyasn1 (0.1.9)
pycparser (2.14)
pyOpenSSL (16.0.0)
requests (2.9.1)
rollbar (0.11.5)
setuptools (20.3.1)
simplegeneric (0.8.1)
six (1.10.0)
traitlets (4.2.1)
Twisted (16.0.0)
zope.interface (4.1.3)
```

It does not matter if I wait until reactor starts (eg. with reactor.callLater()), calling rollbar.report_message() always ends with above error.

**Comment by zgoda-mobica** [comment]
This seems to come from imported unicode_literals. And there's more problems with this setting there:

```
Unhandled error in Deferred:
Traceback (most recent call last):
Failure: twisted.web._newclient.RequestGenerationFailed: [<twisted.python.failure.Failure exceptions.TypeError: Must write bytes to a TLS transport, not unicode.>]
```

**Comment by zgoda-mobica** [comment]
PR https://github.com/rollbar/pyrollbar/pull/106 fixes this issue.

**Comment by ezarowny** [comment]
Confirmed reproducible infinite loop.

**Comment by ezarowny** [comment]
Addressed in #109
